src/common/type/keyValue.c

```diff
@@ -331,3 +331,39 @@ kvGetList(const KeyValue *const this, const Variant *const key)
 
     FUNCTION_TEST_RETURN(VARIANT_LIST, result);
 }
+
+/**********************************************************************************************************************************/
+KeyValue *
+kvRemove(KeyValue *this, const Variant *key)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(KEY_VALUE, this);
+        FUNCTION_TEST_PARAM(VARIANT, key);
+    FUNCTION_TEST_END();
+
+    ASSERT(this != NULL);
+    ASSERT(key != NULL);
+
+    MEM_CONTEXT_BEGIN(objMemContext(this))
+    {
+        // Find the key
+        unsigned int listIdx = kvGetIdx(this, key);
+
+        // If the key was found, remove it
+        if (listIdx != KEY_NOT_FOUND)
+        {
+            // Remove from the list
+            lstRemoveIdx(this->list, listIdx);
+
+            // Remove from the key list
+            for (unsigned int idx = 0; idx < varLstSize(this->pub.keyList); idx++)
+            {
+                if (varEq(key, varLstGet(this->pub.keyList, idx)))
+                    lstRemoveIdx((List *)this->pub.keyList, idx);
```

Comment:
We should free the key/value being removed here or we'll leak memory.

Reply:
Hm, I'm not sure to exactly know how to do that (I thought lstRemoveIdx would handle it) but I tried something in 20d3108

Reply:
Looks better, but I did some more cleanup in d65796423f1f9d.

Reply:
oh I see the missing varFree now. Thanks for the update, it looks better indeed!